Distribution/Server/Features/Html.hs

```diff
@@ -1474,37 +1521,70 @@ mkHtmlTags HtmlUtilities{..}
         tagList <- queryGetTagList
         let withCounts = filter ((>0) . snd) . map (\(tg, pkgs) -> (tg, Set.size pkgs)) $ tagList
             countSort = sortBy (flip compare `on` snd) withCounts
-        return $ toResponse $ Resource.XHtml $ hackagePage "Hackage tags" $
+        return $ toResponse $ Resource.XHtml $ hackagePage "Hackage tags"
           [ h2 << "Hackage tags"
           , h4 << "By name"
-          , paragraph ! [theclass "toc"] << (intersperse (toHtml ", ") $ map (tagItem . fst) withCounts)
+          , paragraph ! [theclass "toc"] << intersperse (toHtml ", ") (map (tagItem . fst) withCounts)
           , h4 << "By frequency"
-          , paragraph ! [theclass "toc"] << (intersperse (toHtml ", ") $ map (toHtml . tagCountItem) countSort)
+          , paragraph ! [theclass "toc"] << intersperse (toHtml ", ") (map (toHtml . tagCountItem) countSort)
           ]
       where tagCountItem (tg, count) =
               [ tagItem tg
               , toHtml $ " (" ++ show count ++ ")"
               ]
             tagItem tg = anchor ! [href $ tagUri tags "" tg] << display tg
 
+    putAliasEdit :: DynamicPath -> ServerPartE Response
+    putAliasEdit dpath = do
+        tagname <- tagInPath dpath
+        mergeTags (Tag tagname)
```

Comment:
This `mergeTags` isn't at the right abstraction level. It includes front-end things like looking up query params and then performing some state update. Instead all the front-end bits should live here and the code in the underlying feature should just deal with validation (if any) and the state/cache updates.

We can fix this by moving parts of the `mergeTags` code out to here.